linera-core/src/unit_tests/client_tests.rs

```diff
@@ -1643,76 +1680,1188 @@ where
+        .block()
+        .unwrap()
+        .operations
+        .contains(&Operation::System(SystemOperation::Transfer {
+            owner: None,
+            recipient: Recipient::Burn,
+            amount: Amount::from_tokens(1),
+            user_data: UserData::default(),
+        })));
+
+    // Previous should be the `ChangeOwnership` operation, as the blob operations shouldn't be executed here.
+    assert!(hashed_certificate_values[1]
+        .inner()
+        .block()
+        .unwrap()
+        .operations
+        .contains(&owner_change_op));
+    Ok(())
+}
+
+#[test_case(MemoryStorageBuilder::default(); "memory")]
+#[cfg_attr(feature = "storage-service", test_case(ServiceStorageBuilder::new().await; "storage_service"))]
+#[cfg_attr(feature = "rocksdb", test_case(RocksDbStorageBuilder::new().await; "rocks_db"))]
+#[cfg_attr(feature = "dynamodb", test_case(DynamoDbStorageBuilder::default(); "dynamo_db"))]
+#[cfg_attr(feature = "scylladb", test_case(ScyllaDbStorageBuilder::default(); "scylla_db"))]
+#[test_log::test(tokio::test)]
+async fn test_blob3<B>(storage_builder: B) -> anyhow::Result<()>
+where
+    B: StorageBuilder,
+    ViewError: From<<B::Storage as Storage>::StoreError>,
+{
+    // Configure a chain with two regular and no super owners.
+    let mut builder = TestBuilder::new(storage_builder, 4, 0).await?;
+    let description1 = ChainDescription::Root(1);
+    let description2 = ChainDescription::Root(2);
+    let description3 = ChainDescription::Root(3);
+    let chain_id3 = ChainId::from(description3);
+    let client0 = builder
+        .add_initial_chain(description1, Amount::ZERO)
+        .await?;
+    let client1 = builder
+        .add_initial_chain(description2, Amount::ZERO)
+        .await?;
+    let mut client2 = builder
+        .add_initial_chain(description3, Amount::from_tokens(10))
+        .await?;
+    let pub_key2 = client2.public_key().await.unwrap();
+    let key_pair3 = KeyPair::generate();
+    let pub_key3 = key_pair3.public();
+    let key_pair4 = KeyPair::generate();
+    let pub_key4 = key_pair4.public();
+    let owner_change_op = Operation::System(SystemOperation::ChangeOwnership {
+        super_owners: Vec::new(),
+        owners: vec![(pub_key2, 50), (pub_key3, 50), (pub_key4, 50)],
+        multi_leader_rounds: 10,
+        timeout_config: TimeoutConfig {
+            fast_round_duration: Some(TimeDelta::from_secs(5)),
+            ..TimeoutConfig::default()
+        },
+    });
+    client2
+        .execute_operation(owner_change_op.clone())
+        .await
+        .unwrap();
+    let mut client3 = builder
+        .make_client(
+            chain_id3,
+            key_pair3,
+            client2.block_hash(),
+            BlockHeight::from(1),
+        )
+        .await?;
+    let client4 = builder
+        .make_client(
+            chain_id3,
+            key_pair4,
+            client2.block_hash(),
+            BlockHeight::from(1),
+        )
+        .await?;
```

Comment:
So in summary:
* Chain 1 is owned by client 0.
* Chain 2 is owned by client 1.
* Chain 3 is owned by clients 2, 3 and 4? :dizzy_face:

Reply:
Ha yeah, sorry, that's pretty confusing. I'll do better names (more related to the chains)